### PR TITLE
Events Dms Quotes

### PR DESCRIPTION
**Pull request by peytono**
- Chats
  - auto scroll to end of messages
  - bug fix: user can open a dm, if no existing dms
  - remove lines between messages
  - change input to textarea; resizes when message gets long and message is sent
  - dm username no longer wraps
- Events
  - gallery only shows current events
  - calendar still shows past events; if clicked brings you to that past event's page
  - created an unused custom agenda component; currently is an incompatible type for calendar
  - Event Page 
    - address click opens an external link to that address on google maps
    - update event page styling Screenshot 2024-07-11 at 3 52 50 PM
    - image modal blurs background
    - bug fix: logo renders on refresh
- Quotes
  - updated button placement; responsive to mobile vs desktop
Screenshot 2024-07-11 at 3 49 12 PMScreenshot 2024-07-11 at 3 49 26 PM
  - bug fix: logic to fix api occassionally sending entire catergory of quotes in request

### CODE REVIEW

File: src/components/Chat.tsx [hunk 1]
```diff
@@ -114,12 +115,12 @@ function Chat() {
     }
   };
 
-  useEffect(() => {
-    const addMessage = (msg: string, clientOffset: string) => {
-      setMessages((curMessages) => curMessages.concat([{ msg, clientOffset }]));
-    };
-    socket.on('sendMsg', addMessage);
-  }, [setMessages]);
+  // useEffect(() => {
```
Comment (kycodee) [inline src/components/Chat.tsx:118]: Is this useEffect still needed?

File: src/components/Chat.tsx [hunk 2]
```diff
@@ -139,21 +140,32 @@ function Chat() {
...
     socket.on('sendDm', addDm);
   }, [setDms]);
 
-  const onSend = () => {
-    if (message) {
-      socket.emit('msg', message, socket.id);
-    }
-    setMessage('');
-  };
+  // const onSend = () => {
```
Comment (kycodee) [inline src/components/Chat.tsx:163]: This onSend function is commented out, but is being called down below.

File: src/components/Chat.tsx [hunk 3]
```diff
@@ -170,9 +182,10 @@ function Chat() {
     const { id } = e.target;
     const { key } = e;
     if (key === 'Enter') {
-      if (id === 'chat') {
-        onSend();
-      } else if (id === 'user') {
+      // if (id === 'chat') {
```
Comment (kycodee) [inline src/components/Chat.tsx:185]: Is this commented out conditional still necessary?

File: src/components/ChatBot.tsx [hunk 4]
```diff
@@ -2,7 +2,7 @@ import React, { useState, useEffect, useContext, useRef } from 'react';
 import {
   Heading,
   Center,
-  StackDivider,
+  // StackDivider,
```
Comment (kycodee) [inline src/components/ChatBot.tsx:5]: StackDivider is no longer being imported, is it still needed in this file?

File: src/components/ChatComponents/ChatInput.tsx [hunk 5]
```diff
@@ -1,33 +1,54 @@
...
 }) {
+  useEffect(() => {
+    const curTextareaRef = textareaRef.current;
+    autosize(curTextareaRef);
+    return () => {
+      autosize.destroy(curTextareaRef);
+    };
+  }, [textareaRef]);
+
+  // useEffect(() => {
```
Comment (kycodee) [inline src/components/ChatComponents/ChatInput.tsx:37]: Will this useEffect be used again in the future or can it be deleted?

**Review by kycodee — CHANGES_REQUESTED** [review]
Really good and clean work!

File: src/components/Chat.tsx (hunk 2 above)
Comment (peytono) [reply src/components/Chat.tsx:163]: onSendDm is being called elsewhere, but not onSend

File: src/components/ChatBot.tsx (hunk 4 above)
Comment (peytono) [reply src/components/ChatBot.tsx:5]: I can go ahead and remove this unused import

File: src/components/ChatComponents/ChatInput.tsx (hunk 5 above)
Comment (peytono) [reply src/components/ChatComponents/ChatInput.tsx:37]: I'll go ahead and delete it, ended up moving this logic elsewhere

File: src/components/Chat.tsx (hunk 1 above)
Comment (peytono) [reply src/components/Chat.tsx:118]: This is along with other commented out code for this file is logic for the global dms. Wasn't sure if I should save it incase we wanted to add it back later

File: src/components/Chat.tsx (hunk 2 above)
Comment (kycodee) [reply src/components/Chat.tsx:163]: Sorry about that, you're right. I was looking at the deleted part it was being used in.

File: src/components/ChatBot.tsx (hunk 4 above)
Comment (kycodee) [reply src/components/ChatBot.tsx:5]: Sounds good.

File: src/components/ChatComponents/ChatInput.tsx (hunk 5 above)
Comment (kycodee) [reply src/components/ChatComponents/ChatInput.tsx:37]: All good.

**Pull request merged**